bgpd/bgp_route.c

```diff
@@ -4515,7 +4515,8 @@ int bgp_nlri_parse_ip(struct peer *peer, struct attr *attr,
 			if (pnt + BGP_ADDPATH_ID_LEN >= lim)
 				return BGP_NLRI_PARSE_ERROR_PACKET_OVERFLOW;
 
-			addpath_id = ntohl(*((uint32_t *)pnt));
+			memcpy(&addpath_id, pnt, 4);
```

Comment:
Minor nit : I think instead of using 4 we can use sizeof(uint32_t) or BGP_ADDPATH_ID_LEN macro which is 4.